### ISSUE TITLE
Multi-Instance Task: Parallel Execution Issue with Data Store Inputs

### ISSUE THREAD

**Issue by usama9500** [sartography/SpiffWorkflow#411]
Multi-instance tasks, designed to execute in parallel, are instead running sequentially when pulling input data from a data store.

**Comment by essweine** [comment]
This is a duplicate of a documentation ticket in spiff-arena https://github.com/sartography/spiff-arena/issues/1572.

Note that this is not an accurate description of the problem, which was that one of our users had to use a sequential multiinstance instead of a parallel multiinstance to get the model working.
